### ISSUE TITLE
Persistent scroll position with TabBatView

### ISSUE THREAD

**Issue by chrislrobson** [toshiaki-h/split_view#4]
Problem:  If SplitView() is used within a TabBarView(and maybe even other widget) the scroll position within a view port is not maintained but reverts back to the "initialWeight" position.  While this might have been the intended behavior it does not provide the best user experience.
To provide persistent scroll position the following changes are suggested:

Create the following factory below and change the following lines to access the new changed "defaultWeight" variable

Change line 41 to
svd.defaultWeight = widget.initialWeight;

Change lines 73/74 to:
weight = ValueNotifier(svd.defaultWeight);
    _prevWeight = svd.defaultWeight;

Add line 80 after the "builder:" statement 
svd.defaultWeight = w;

class SplitViewData {
  static final SplitViewData _splitviewdata = SplitViewData._internal();
  double defaultWeight;
  factory SplitViewData() {
    return _splitviewdata;
  }
  SplitViewData._internal();
}
final svd = SplitViewData();

**Comment by toshiaki-h** [comment]
Thank you for your suggestion.
I'm now implementing your idea in the library.
Please wait a few days.

**Comment by toshiaki-h** [comment]
@chrislrobson I tried your code. But it didn't work well when I had multiple split views.
So I adopted the method of using PageStorageKey and tried to implement your idea. I haven't posted it on pub.dev yet, but I've posted the latest code on github so can you check it out?

**Comment by chrislrobson** [comment]
Yep, that makes a lot of sense.  I'm only using a single split screen, 
sorry I didnt test that.

The good news is you've found an even better solution.  I'll look at it, 
in fact there is another issue I was thinking PageStorageKey might fix 
when using TabbarViews.  When you leave a tab then come back to it, the 
scroll position is reset.  Not only does my fix only works for a single 
split screen but doesnt overcome the switching tabs problem.

by-the-way, if your interested I just released a new version of my app 
if you want to see how and why I use the split screen. Namely it allows 
Amateur Radio operators to view what we call "check-ins" on one split 
view and listen to an audio from websites that broadcast the associated 
radio signal on the other view.  It cannot do this with tab views 
because the webpages audio wont keep broadcasting but since the split 
views are still active displays the audio is maintained.  My Amateur 
Radio friends love it.

Anyway, the App name is "On-Air Net Scraper" and is at both Apple and 
Google stores if curious.

thanks for the update.....73....Chris

**Comment by chrislrobson** [comment]
Toshiaki

Wanted to let you know I'm testing code today, be most of the day since 
I'll be integrating a 3-way split into my app.

let you know how it turns out l8r today.....

Chris

**Comment by chrislrobson** [comment]
Toshiaki

Unfortunately, the use of PageStorageKey with 2-way-slit view within a 
TabBarView still doesn't work.  When the view updates, which it must do 
every 30 seconds to display new data, the view resets to the default top 
position scroll.  Ergo, I have to go back to my code modifications.

I'm going to see what I can come up with for the 3-way-split issue.

Chris

**Comment by toshiaki-h** [comment]
Chris

Thank you for the test. I will check it too.

Toshiaki

**Comment by chrislrobson** [comment]
Ok, Toshiaki, got it working with just a few minor changes to my code.

Hopefully by EOB I'll have a cleaned up version.

It still resets when you change tabs but it maintains scroll position as 
the 2-way did as long as your viewing the active views.

I'm going to see if there is a fix for leaving the current active tab 
view but to be honest I'm not sure that is possible or at least not as 
easy a fix, aka I'm sure its going to take some global position variable 
saves to accomplish.

l8r....Chris

**Comment by toshiaki-h** [comment]
Chris

I added identifier field read/writeState methods. In order to identify the weight each SplitView.
Sample program (main_tabview.dart) seems to be working properly, could you check that?
